libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +737,96 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionInfo> baseFuncs;
+
+	for (ContractDefinition const* c: resolveBaseContracts(_contract))
+		for (pair<string, FunctionInfo> const& funcPair: getOverridableFunctions(c))
+		{
+			auto baseFunc = funcPair.second.funcDef;
+
+			auto baseSearchResult =
+				find_if(
+					baseFuncs.cbegin(),
+					baseFuncs.cend(),
+					[&] (auto const& f) {
+						return hasEqualNameAndParameters(*(f.funcDef), *baseFunc);
+					}
+				);
+
+			if (baseSearchResult == baseFuncs.cend())
+			{
+				baseFuncs.emplace_back(funcPair.second);
+				continue;
+			}
+			else if (baseSearchResult->contract == funcPair.second.contract)
+				continue;
+
+			auto const& defFuncs = _contract.definedFunctions();
+			auto const derivedSearchresult = find_if(
+				defFuncs.cbegin(),
+				defFuncs.cend(),
+				[&] (auto const& f) {
+					return hasEqualNameAndParameters(*f, *baseFunc);
+				}
+			);
+
+			if (derivedSearchresult != defFuncs.cend())
+				continue;
+
+			SecondarySourceLocation ssl;
+			ssl.append("First definition here: ", baseSearchResult->funcDef->location());
+			ssl.append("Second definition here: ", baseFunc->location());
+
+			m_errorReporter.typeError(
+				_contract.location(),
+				ssl,
+				"Ambiguous base function " +
+				baseFunc->name() +
+				"() must be overridden."
+			);
+		}
+}
+
+vector<ContractDefinition const*> ContractLevelChecker::resolveOverrideList(OverrideSpecifier const& _overrides) const
+{
+	std::vector<ASTPointer<UserDefinedTypeName>> const& overrides = _overrides.overrides();
+
+	vector<ContractDefinition const*> resolved{overrides.size(), nullptr};
+
+	transform(
```

Comment:
I have the felling that a for each loop would lead to less clutter and could even be shorter.